usecases/ast_eval/evaluate_ast_expression.go

```diff
@@ -29,12 +29,12 @@ func (evaluator *EvaluateAstExpression) EvaluateAstExpression(
 
 	evaluation, ok := EvaluateAst(ctx, environment, ruleAstExpression)
 	if !ok {
-		return ast.RootNodeEvaluation{}, errors.Join(evaluation.AllErrors()...)
+		return false, evaluation, errors.Join(evaluation.AllErrors()...)
 	}
 
-	rootEvaluation, err := ast.AdaptRootNodeEvaluation(evaluation)
+	returnValue, err := evaluation.GetBoolReturnValue()
 	if err != nil {
-		return ast.RootNodeEvaluation{}, errors.Join(models.ErrRuntimeExpression, err)
+		return false, evaluation, errors.Join(models.ErrRuntimeExpression, err)
 	}
-	return rootEvaluation, nil
+	return returnValue, evaluation, nil
```

Comment:
Now return `(bool, ast.NodeEvaluation, error)` where.

`evaluation` is always returned (so we can store intermediate calc info to get context on errors)